weldx/transformations.py

```diff
@@ -379,45 +518,449 @@ def _calculate_orthogonal_axis(a_0, a_1):
         return np.cross(a_0, a_1)
 
     @property
-    def basis(self):
+    def basis(self) -> xr.DataArray:
         """
         Get the normalizes basis as matrix of 3 column vectors.
 
         This function is identical to the 'orientation' function.
 
         :return: Basis of the coordinate system
         """
-        return self._orientation
+        return self._dataset.basis.transpose(..., "c", "v")
 
     @property
-    def orientation(self):
+    def orientation(self) -> xr.DataArray:
         """
         Get the coordinate systems orientation matrix.
 
         This function is identical to the 'basis' function.
 
         :return: Orientation matrix
         """
-        return self._orientation
+        return self._dataset.basis.transpose(..., "c", "v")
 
     @property
-    def origin(self):
+    def origin(self) -> xr.DataArray:
         """
         Get the coordinate systems origin.
 
         This function is identical to the 'location' function.
 
         :return: Origin of the coordinate system
         """
-        return self._location
+        return self._dataset.origin.transpose(..., "c")
 
     @property
-    def location(self):
+    def location(self) -> xr.DataArray:
         """
         Get the coordinate systems location.
 
         This function is identical to the 'origin' function.
 
         :return: Location of the coordinate system.
         """
-        return self._location
+        return self._dataset.origin.transpose(..., "c")
+
+    @property
+    def time(self) -> pd.DatetimeIndex:
+        """
+        Get the time union of the local coordinate system (or None if system is static).
+
+        :return: DateTimeIndex-like time union
+        """
+
+        if "time" in self._dataset.coords:
+            return pd.DatetimeIndex(self._dataset.time.data)
+        return None
+
+    @property
+    def dataset(self) -> xr.Dataset:
+        """
+        Get the underlying xarray Dataset.
+
+        :return: xarray Dataset with origin and basis as DataVariables.
+        """
+        return self._dataset
+
+    def interp_time(self, time: pd.DatetimeIndex) -> "LocalCoordinateSystem":
+        """
+        Interpolates the data in time.
+
+        :param time: Series of times.
+        :return: Coordinate system with interpolated data
+        """
+        if not isinstance(time, pd.DatetimeIndex):
+            raise Exception("Invalid parameter type.")
```

Comment:
Instead of raising an exception, maybe we should allow for more flexibility by trying to cast to `pd.DatetimeIndex` here.
I think `List[pd.Timestamp]` (allready works) or a single `pd.Timestamp` (works if converted to List) should also be allowed and listed in type hints.